lib/file_store/s3_store.rb

```diff
@@ -391,14 +437,17 @@ def get_upload_key(upload)
       end
     end
 
-    def update_ACL(key, secure)
-      begin
-        object_from_path(key).acl.put(
-          acl: SiteSetting.s3_use_acls ? (secure ? "private" : "public-read") : nil,
-        )
-      rescue Aws::S3::Errors::NoSuchKey
-        Rails.logger.warn("Could not update ACL on upload with key: '#{key}'. Upload is missing.")
+    def update_access_control(key, secure)
+      object = object_from_path(key).acl.put(acl: acl_option_value(secure:))
```

Comment:
Should we be surrounding this with a check for `SiteSetting.s3_use_acls`? Maybe its already checked elsewhere though

Reply:
Yup this is now checked in `acl_option_value` with the logic centralized in a single method as well.

https://github.com/discourse/discourse/pull/32988/files#diff-8ea7a76a0c3006ff740b3d346d09a995b2028c3ad47dd2c3032ff67381ab414aR368-R371